fl4health/clients/deep_mmd_clients/ditto_deep_mmd_client.py

```diff
@@ -0,0 +1,201 @@
+from logging import ERROR
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, Scalar
+
+from fl4health.checkpointing.client_module import CheckpointMode, ClientCheckpointModule
+from fl4health.clients.basic_client import TorchInputType
+from fl4health.clients.ditto_client import DittoClient
+from fl4health.losses.deep_mmd_loss import DeepMmdLoss
+from fl4health.model_bases.feature_extractor_buffer import FeatureExtractorBuffer
+from fl4health.utils.losses import EvaluationLosses, LossMeterType
+from fl4health.utils.metrics import Metric
+
+
+class DittoDeepMmdClient(DittoClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[ClientCheckpointModule] = None,
+        lam: float = 1.0,
+        deep_mmd_loss_weight: float = 10.0,
+        flatten_feature_extraction_layers: Dict[str, bool] = {},
+        size_feature_extraction_layers: Dict[str, int] = {},
```

Comment:
We noted that we don't want mutable defaults in the previous PR. Just wanted to put this here as a reminder to change these around too 🙂